x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/update_source_editor.js

```diff
@@ -287,9 +290,83 @@ export class UpdateSourceEditor extends Component {
             compressed
           />
         </EuiFormRow>
+      </EuiPanel>
+    );
+  }
+
+  _renderScalingPanel() {
+    const scalingOptions = [
+      {
+        id: SCALING_TYPES.LIMIT,
+        label: i18n.translate('xpack.maps.source.esSearch.limitScalingLabel', {
+          defaultMessage: 'Limit results to {maxResultWindow}.',
+          values: { maxResultWindow: this.state.maxResultWindow },
+        }),
+      },
+      {
+        id: SCALING_TYPES.TOP_HITS,
+        label: i18n.translate('xpack.maps.source.esSearch.useTopHitsLabel', {
+          defaultMessage: 'Show top hits per entity.',
+        }),
+      },
+    ];
+    if (this.state.supportsClustering) {
+      scalingOptions.push({
+        id: SCALING_TYPES.CLUSTERS,
+        label: i18n.translate('xpack.maps.source.esSearch.clusterScalingLabel', {
+          defaultMessage: 'Show clusters when results exceed {maxResultWindow}.',
+          values: { maxResultWindow: this.state.maxResultWindow },
+        }),
+      });
+    }
 
-        <EuiHorizontalRule margin="xs" />
-        {this.renderTopHitsForm()}
+    let filterByBoundsSwitch;
+    if (this.props.scalingType !== SCALING_TYPES.CLUSTERS) {
+      filterByBoundsSwitch = (
+        <EuiFormRow>
+          <EuiSwitch
+            label={i18n.translate('xpack.maps.source.esSearch.extentFilterLabel', {
+              defaultMessage: 'Dynamically filter for data in the visible map area',
+            })}
+            checked={this.props.filterByMapBounds}
+            onChange={this._onFilterByMapBoundsChange}
+            compressed
+          />
+        </EuiFormRow>
+      );
+    }
+
+    let scalingForm = null;
+    if (this.props.scalingType === SCALING_TYPES.TOP_HITS) {
+      scalingForm = (
+        <Fragment>
+          <EuiHorizontalRule margin="xs" />
+          {this._renderTopHitsForm()}
+        </Fragment>
+      );
+    }
+
+    return (
+      <EuiPanel>
+        <EuiTitle size="xs">
+          <h5>
+            <FormattedMessage id="xpack.maps.esSearch.scaleTitle" defaultMessage="Scaling" />
```

Comment:
this is a nit, but I don't think we should call this "Scaling". It;s a little too opaque.

Maybe something like:
- "Query strategy"
- "Limit document number"
- ...

Something that hints better that there will be some sort of data-reduction going on. 

There's the following:
- omission: Limit data to first 10k results
- select last known: Select only last n documents by entity (aka top hits)
- clustering: cluster many documents in clusters

fwiw, we don't need to derail this PR in a bike-shedding discussion about naming, but I feel like we can make an incremental improvement, as things are starting to get more complex. the dumberer the language the better I suppose.

Reply:
I am open to changing the wording but would like to get the copy settled before changing things over and over again. 

I am hesitant to call top hots to `select last known` since the top hits is effected by sorting and does not always show the last known. If sorting is ascending then its the first "known". And if sorting is not on a time field then its not "first/last known" at all.

Reply:
as discussed, let's move this to separate PR